02_bubble/patrickmarabeas/main.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+// Bubble implements the Bubble sort algorithm
+func bubble(s []int) []int {
+	length := len(s)
+	// Create copy
+	result := make([]int, length)
+	copy(result, s)
```

Comment:
This forces a copy that the caller may not require. Making a copy should be left up to the caller.

Reply:
This is a good point, thanks. I'll make sure I'm not being opinionated with immutability in the future.